snow/consensus/snowman/poll/graph.go

```diff
@@ -61,54 +56,48 @@ func buildVoteGraph(getParent func(ids.ID) (ids.ID, bool), votes bag.Bag[ids.ID]
 			v.parent = u
 			u.descendants = append(u.descendants, v)
 		} else {
+			// This case is typically dead code, as it can only happen if
+			// getParent returns new mappings that were not returned earlier in
+			// the loop. But it is handled for completeness.
 			roots = append(roots, v)
 		}
 	}
 
 	leaves := findLeaves(id2Vertex)
-
-	return voteGraph{leaves: leaves, roots: roots, vertexCount: len(id2Vertex)}
-}
-
-func addAncestorsToGraph(getParent func(ids.ID) (ids.ID, bool), idList []ids.ID, id2Vertex map[ids.ID]*voteVertex) {
-	var discoveredAncestors bag.Bag[ids.ID]
-
-	for _, id := range idList {
-		ancestors := discoverAncestorsOfID(getParent, id, id2Vertex)
-		discoveredAncestors.Add(ancestors.List()...)
-	}
-
-	for _, id := range discoveredAncestors.List() {
-		id2Vertex[id] = &voteVertex{id: id}
```

Comment:
This function seemed unnecessary to me. We are doing quite a bit more work adding all these entries into bags (which can cause unneeded duplicate traversals). Rather than allocating bags we can just make the entries in the graph immediately.